frontend/amundsen_application/api/metadata/v0.py

```diff
@@ -853,6 +867,190 @@ def get_column_lineage() -> Response:
+def get_resource_description() -> Response:
+    try:
+        resource_type = get_query_param(request.args, 'type')
+        resource_key = get_query_param(request.args, 'key')
+
+        endpoint = _get_endpoint_from_resource_type(resource_type)
+
+        url = '{0}/{1}/description'.format(endpoint, resource_key)
+
+        response = request_metadata(url=url)
+        status_code = response.status_code
+
+        if status_code == HTTPStatus.OK:
+            message = 'Success'
+            description = response.json().get('description')
+        else:
+            message = 'Get resource description failed'
+            description = None
+
+        payload = jsonify({'description': description, 'msg': message})
+        return make_response(payload, status_code)
+    except Exception as e:
+        payload = jsonify({'description': None, 'msg': 'Encountered exception: ' + str(e)})
+        return make_response(payload, HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+@metadata_blueprint.route('/put_resource_description', methods=['PUT'])
+def put_resource_description() -> Response:
+    try:
+        args = request.get_json()
+        resource_type = get_query_param(args, 'type')
+        resource_key = get_query_param(args, 'key')
+        description = get_query_param(args, 'description')
+
+        endpoint = _get_endpoint_from_resource_type(resource_type)
+
+        url = '{0}/{1}/description'.format(endpoint, resource_key)
+
+        response = request_metadata(url=url, method='PUT', data=json.dumps({'description': description}))
+        status_code = response.status_code
+
+        if status_code == HTTPStatus.OK:
+            message = 'Success'
+        else:
+            message = 'Update resource description failed'
+
+        payload = jsonify({'msg': message})
+        return make_response(payload, status_code)
+    except Exception as e:
+        payload = jsonify({'msg': 'Encountered exception: ' + str(e)})
+        return make_response(payload, HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+@metadata_blueprint.route('/get_resource_generation_code', methods=['GET'])
+def get_resource_generation_code() -> Response:
+    """
+    Call metadata service to fetch table generation code
+    :return:
+    """
+    try:
+        resource_type = get_query_param(request.args, 'type')
+        resource_key = get_query_param(request.args, 'key')
+
+        endpoint = _get_endpoint_from_resource_type(resource_type)
+
+        url = f'{endpoint}/{resource_key}/generation_code'
+        response = request_metadata(url=url, method=request.method)
+        payload = response.json()
+        return make_response(jsonify(payload), 200)
+    except Exception as e:
+        payload = jsonify({'msg': 'Encountered exception: ' + str(e)})
+        return make_response(payload, HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+@metadata_blueprint.route('/get_resource_lineage', methods=['GET'])
+def get_resource_lineage() -> Response:
```

Comment:
Is there a subsequent change planned for the static files to use these new generic methods so that the resource specific methods (e.g. `get_table_lineage`, `get_column_lineage`, etc.) can be removed?

Reply:
Ended up changing this because it didn't make sense to do right now, this refactoring needs a bit more thought so just followed the convention from tables